tests/parsing/FilterHelperTest.py

```diff
@@ -0,0 +1,53 @@
+import unittest
+
+from coalib.parsing.FilterHelper import (
+    apply_filter,
+    apply_filters,
+    is_valid_filter,
+    _filter_section_bears
+)
+from coalib.parsing.filters import available_filters
+from coalib.parsing.InvalidFilterException import InvalidFilterException
+from coalib.settings.ConfigurationGathering import get_all_bears
+
+
+class FilterHelperTest(unittest.TestCase):
+
+    def test_apply_filter_exception(self):
+        with self.assertRaises(InvalidFilterException) as exp:
+            apply_filter('unknown', ['args'])
+
+        message = str(exp.exception)
+        self.assertEqual("'unknown' is an invalid filter. Available filters: "
+                         + ', '.join(sorted(available_filters)), message)
+
+    def test_apply_filters_exception(self):
+        with self.assertRaises(InvalidFilterException) as exp:
+            apply_filters([['language', 'C', 'Python'],
+                           ['unknown', 'args']])
+
+        message = str(exp.exception)
+        self.assertEqual("'unknown' is an invalid filter. Available filters: "
+                         + ', '.join(sorted(available_filters)), message)
+
+    def test_is_valid_filter(self):
+        wrong_filter_result = is_valid_filter('check')
+        right_filter_result = is_valid_filter('can_detect')
+        self.assertFalse(wrong_filter_result)
+        self.assertTrue(right_filter_result)
+
+    def test_filter_section_bears(self):
+        local_bears = get_all_bears()[0]
+        filter_args = {'c', 'java'}
+        result_for_filter_section = _filter_section_bears(
+            local_bears, filter_args, 'language')
+        self.assertIsNotNone(result_for_filter_section)
+
+    def test_apply_filter(self):
+        apply_filter_result = apply_filter('language', ['c', 'java'])
+        self.assertIsNotNone(apply_filter_result)
+
+    def test_apply_filters(self):
+        apply_filters_result = apply_filters([['language', 'C', 'Python'],
```

Comment:
whaaa .. here is this array of array again.  who designed this shit.

Reply:
https://github.com/coala/coala/issues/5398 is for fixing that problem, but first lets get this patch fixed and merged.